### ISSUE TITLE
`preventOverrun` maybe not working for CronJob

### ISSUE THREAD

**Issue by boristian** [kibertoad/toad-scheduler#195]
Hi,
first of alle thanks for the great work!
Here is an example where a CronJob an a SimpleIntervalJob are created both with an Interval of 10 seconds and the same task which takes > 17 seconds to execute and both with the option `preventOverrun`.
While the SimpleIntervalJob works and is only started every second time (20 seconds) the Cronjob is called every 10 seconds.
It seems like the option is not working for CronJobs or am i'm doing something wrong?

```js
const {
  AsyncTask,
  CronJob,
  SimpleIntervalJob,
  ToadScheduler
} = require('toad-scheduler')

const delay = (timeout = 2000) => new Promise(resolve => {
  setTimeout(() => resolve(), timeout)
})

const cronTaskFn = () => {
  const now = new Date().toISOString()
  // const now = Date.now()

  console.error(new Date().toISOString(), 'CronTask Start', now)

  return delay(17000)
    .then(() => console.error(new Date().toISOString(), 'CronTask End', now))
}

const simpleTaskFn = () => {
  const now = new Date().toISOString()
  // const now = Date.now()

  console.error(new Date().toISOString(), ': SimpleTask Start', now)

  return delay(17000)
    .then(() => console.error(new Date().toISOString(), 'SimpleTask End', now))
}

const run = () => {
  try {
    const scheduler = new ToadScheduler()

    // Scheduler-Setup
    const cronSchedule = { cronExpression: '*/10 * * * * *' }
    const cronTask = new AsyncTask('cron-task', () => cronTaskFn())
    const cronOpts = { id: 'CRON_TASK_ID', preventOverrun: true }

    console.debug('Scheduler schedule and opts', cronSchedule, cronOpts)

    const cronJob = new CronJob(cronSchedule, cronTask, cronOpts)

    scheduler.addCronJob(cronJob)

    const simpleTask = new AsyncTask('simple-task', () => simpleTaskFn())
    const simpleOpts = { id: 'SIMPLE_TASK_ID', preventOverrun: true }

    console.debug('Scheduler opts', simpleOpts)

    const simpleJob = new SimpleIntervalJob({ seconds: 10 }, simpleTask, simpleOpts)

    scheduler.addSimpleIntervalJob(simpleJob)
  } catch (err) {
    console.error(err)

    process.exit(1)
  }

  process.on('SIGINT', () => process.exit(0))
}

run()
```

I am using version 3.0.0.

Thanks in advance

**Comment by kibertoad** [comment]
@Hexagon

**Comment by Hexagon** [comment]
:wave: 

I tcan not find any code in toad-scheduler that translate option `preventOverrun` to croner option `protect` (https://github.com/Hexagon/croner#options). Could that be the cause?

**Comment by kibertoad** [comment]
@Hexagon is it turned off by default?

**Comment by kibertoad** [comment]
thanks, I'll fix then!

**Comment by ahamid** [comment]
it looks like it's explicitly turned off regardless of the option supplied:

https://github.com/kibertoad/toad-scheduler/blob/6d86d793c2aa4dd530700dd3d2ccc76a4b264b1b/lib/engines/cron/CronJob.ts#L35

```typescript
constructor(schedule: CronSchedule, task: Task | AsyncTask, options: JobOptions = {}) {
  super(options.id)
  this.preventOverrun = false // xxx
  this.schedule = schedule
  this.task = task
}
```

**Comment by kibertoad** [comment]
Fixed in 3.0.1, sorry for taking so long, writing a working test for this was much trickier than expected.
